src/main/scala/com/monsanto/arch/cloudformation/model/simple/Builders.scala

```diff
@@ -564,8 +574,9 @@ trait VPC extends Outputs {
 }
 
 trait AvailabilityZone {
-  case class AZ(zone: Token[String])
-  def withAZ(zone: Token[String])(f: (AZ) => Template): Template = f(AZ(zone)) // no resource to create
+  case class AZ(zone: Option[Token[String]])
+  def withAZ(zone: Token[String])(f: (AZ) => Template): Template = f(AZ(Some(zone))) // no resource to create
+  def withAZ(zone: Option[Token[String]])(f: (AZ) => Template): Template = f(AZ(zone)) // no resource to create
 }
```

Comment:
I don't think we need to change this part.  When we use `withAZ`, we're definitely referencing an AZ.